src/cdi_ndi/config.h.in

```diff
@@ -0,0 +1,23 @@
+/*-----------------------------------*-C-*-----------------------------------*/
+/*!
+ * \file   cdi_ndi/config.h
+ * \brief  CPP defines necessary for the cdi_ndi package.
+ * \note   Copyright (C) 2016-2020 Triad National Security, LLC.
+ *         All rights reserved. */
+/*---------------------------------------------------------------------------*/
+
+#ifndef rtt_cdi_ndi_config_h
+#define rtt_cdi_ndi_config_h
+
+/*---------------------------------------------------------------------------*/
+/* Draco/Cmake build system variables */
+/*---------------------------------------------------------------------------*/
+
+/* NDI_ROOT_DIR location, for accessing gendir.all */
+#cmakedefine NDI_ROOT_DIR "@NDI_ROOT_DIR@"
```

Comment:
What is the behavior if `NDI_ROOT_DIR` is not found in the environment?  That is, what occurs if this file is empty?

Reply:
This is a compilation error. Is the best thing to do to use cmake to ensure that `NDI_ROOT_DIR` is set if `NDI_FOUND` is true? I think this is typical because NDI is really only found through `NDI_ROOT_DIR` but I am not sure if this is totally guaranteed right now.

Reply:
`NDI_ROOT_DIR` will not be set in many circumstances (most systems don't have an `ndi` modulefile).  However, we should be able to make a best guess if `libndi.a` is found.  We might even want to update `FindNDI.cmake` to look for `gendir.all` in nearby directories since it isn't always at `$NDI_ROOT_DIR/share`

Reply:
On systems without ndi, I think we should always have `NDI_FOUND` set to false, and since `cdi_ndi/CMakeLists.txt` starts off with a 
`if( NDI_FOUND AND NOT DEFINED ENV{TRAVIS} )`,
I think this will safely prevent `cdi_ndi` from being configured/compiled?
On systems with ndi but no `ndi` modulefile, my understanding is the only way to configure with ndi is to pass a `DNDI_ROOT_DIR` environment variable to cmake. 
Also, I think NDI also exports an environment variable `NDI_GENDIR_PATH` which I think we could just rely on instead of creating our own from `NDI_ROOT_DIR`, or maybe switch from ${NDI_ROOT_DIR}/share/gendir.all if `NDI_GENDIR_PATH` exists.

Reply:
Can you add something like this to `cdi_ndi/CMakeLists.txt` (inside of the `if(NDI_FOUND)` conditional):

```cmake
if( NOT DEFINED ENV{NDI_ROOT_DIR})
  message(FATAL_ERROR "NDI_ROOT_DIR not set. cdi_ndi requires the environment variable NDI_ROOT_DIR to be set if the ndi library is found.")
endif()
```

Reply:
I did the below because `ENV{NDI_ROOT_DIR}` doesn't detect the `-DNDI_ROOT_DIR=` command line argument which also works for `cdi_ndi`:

```cmake
if( NOT DEFINED NDI_ROOT_DIR )                                                   
  message(FATAL_ERROR "NDI_ROOT_DIR not set. cdi_ndi requires the variable \     
          NDI_ROOT_DIR to be set in the environment or with the -DNDI_ROOT_DIR \ 
          flag if the ndi library is found.")                                    
endif()
```
If this is not OK please let me know

Reply:
To clarifiy: when you use `cmake -DFOO=bar ...`, `FOO` is a cmake internal variable, not an environment variable.  When you do `FOO=bar cmake ...`, `FOO` is an environment variable and not a cmake variable.  Maybe this solution is more complete:

```cmake
if( NOT DEFINED NDI_ROOT_DIR AND DEFINED $ENV{NDI_ROOT_DIR})
   set(NDI_ROOT_DIR $ENV{NDI_ROOT_DIR})
endif()
if( NOT DEFINED NDI_ROOT_DIR )                                                   
  message(FATAL_ERROR "NDI_ROOT_DIR not set. cdi_ndi requires the variable "     
          "NDI_ROOT_DIR to be set in the environment or with the -DNDI_ROOT_DIR " 
          "flag if the ndi library is found.")                                    
endif()
```

Reply:
OK this makes sense, done